packages/blade/src/components/List/ListItem.tsx

```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components';
+import React from 'react';
+import { Text } from '../Typography';
+import type { IconComponent } from '../Icons';
+import { useTheme } from '../BladeProvider';
+import { useListContext } from './ListContext';
+import { UnorderedItemIcon } from './ListItemIcons';
+import { ListItemElement } from './ListItemElement';
+import {
+  listItemBulletPaddingRight,
+  listItemBulletMarginTop,
+  listItemOrderedBulletBoxSize,
+  listItemPaddingBottom,
+  listItemPaddingLeft,
+} from './listTokens';
+import type { ListProps } from './List';
+import { getOrderedListItemBullet } from './getOrderedListItemBullet';
+import Box from '~components/Box';
+import { getIn, isValidAllowedChildren, metaAttribute, MetaConstants } from '~utils';
+
+type ListItemProps = {
+  /**
+   * Children to be rendered for ListItem. This can be a text, ListItemLink or another List.
+   *
+   */
+  children: React.ReactNode;
+  /**
+   * Icon to be rendered for a ListItem's bullet.
+   *
+   */
+  icon?: IconComponent;
+  /**
+   * This is a private prop to be used only for internal logic purposes.
+   *
+   */
+  _itemNumber?: undefined;
```

Comment:
i'm afraid people might abuse it 😢 . can't we set this prop as context value or something so that can be read just internally?

Reply:
Couldn't really see a cleaner way to do this than setting a private prop. Context is a bit messy because there will be 0 to 3 levels of ListContext. This makes the logic much simpler and easy to understand